javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/IndyBootstrap.java

```diff
@@ -92,6 +92,12 @@ public class IndyBootstrap {
 
       IndyBootstrapDispatcher.init(
           MethodHandles.lookup().findStatic(IndyBootstrap.class, "bootstrap", bootstrapMethodType));
+
+      // Ensure that CallDepth is already loaded in case of bootstrapAdvice recursions with
+      // ClassLoader.loadClass
+      // This is required because CallDepth is a bootstrap class and therefore triggers our
+      // ClassLoader.loadClass instrumentations
+      Class.forName(CallDepth.class.getName());
```

Comment:
[minor] maybe add that this class needs to be explicitly initialized early because it's the one preventing recursive calls and is thus the one called first when `loadClass` advice method is executed. Adding a comment where this first call is made would also be relevant too just in case someone tries to add something before it.

Also, could we make this static init part of the advice static init instead ? If possible that would be easier to link with the advice method code. On the other hand this `CallDepth` init call might be needed elsewhere.

Reply:
It is in fact not the case that this class needs to be initialized early. It instead needs to be linked to `IndyBootstrap` early:
When some `IndyBootstrap.bootstrapAdvice` is executed for the first time, the JVM encounters a call to `CallDepth.forClass`. When linking this call, the JVM itself invokes `AgentClassLoader.loadClass(CallDepth)` to link that callsite. This in turn will delegate to the instrumented `ClassLoader.loadClass` because it is a bootstrap class. It doesn't matter where within `bootstrapAdvice` this call happens, it just happened by chance that it is the first one.

> Also, could we make this static init part of the advice static init instead

For the reasons above: No, this is not possible, because it wouldn't link the usage of `CallDepth` from within the `IndyBootstrap` class.